sphinx_immaterial/__init__.py

```diff
@@ -130,7 +129,12 @@ def onerror(filename: str, error: Exception) -> None:
                             )
                         excluded = sphinx.util.matching.Matcher(excluded_list)
                     else:
-                        excluded = sphinx.util.matching.DOTFILES
+                        excluded = sphinx.util.matching.Matcher(
```

Comment:
If someone inherits from this theme, then anything matching stylesheets will also be excluded.  Instead, we could modify the build script to place the stylesheets outside of the `static` directory.

Reply:
What location did you have in mind? Reverting this change will go back to copying everything except paths that begin with `.`. So, maybe have the build script put them in a `.theme_css` folder?

Reply:
Perhaps instead of `sphinx_immaterial/static`, the css could be written to `sphinx_immaterial/bundles` -- we might eventually want to do something similar for the bundled javascript.

Reply:
Just so I understand this correctly, you still want the built CSS bundles copied to `outdir` even though they aren't referred to by the base.html template anymore? I'm a little unclear about what "inherits from this theme" means.

Reply:
Sphinx theme's have an "inheritance" mechanism --- the `inherit` field in `theme.conf`.  Our theme inherits from `basic`, although perhaps we should take that out since I think in practice we don't actually use anything from `basic`.

I think inheriting just means the static files, and the HTML templates, are inherited.

I don't know if currently there are any themes that inherit from sphinx-immaterial, but in the future there might be.

No, I don't want the original built CSS bundles  to be copied to the outdir even though they won't be used --- I was just proposing that we change our build script to output them to somewhere other than `static`, so that they don't get copied automatically.  Then we don't have to change the logic here.

Reply:
Ah yes, I had forgotten about inheriting from other sphinx themes like that. I think we could get rid of the basic theme dependency and reduce the logic in this function a bit more.

If you don't want them copied and keep the old logic here, then they should be placed in a folder that begins with `.`. I'll play around a bit, but I think that (using the old logic) wherever we put the bundles in the theme distribution, they will get copied to the same relative location in outdir. The old logic ignored any distribution files whose path begins with `.`, that's why the icons aren't copied to the outdir.

Reply:
>  Our theme inherits from basic, although perhaps we should take that out since I think in practice we don't actually use anything from basic.

I looked into this. There is a way to not inherit from any sphinx thene, but there are some templates that we're using.

1. [page.html](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/themes/basic/page.html) corresponds to the default value for the `templatename` param in [`StandaloneHTMLBuilder.handle_page()`](https://github.com/sphinx-doc/sphinx/blob/cd3f2e435000835dd98a11497440bc16a79ec31c/sphinx/builders/html/__init__.py#L1021). I was able to override this, but it led to other templates that were missing.
2. [genindex.html](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/themes/basic/genindex.html) - This is where I stopped looking into not inheriting from the basic theme. If it involves creating templates specific to Sphinx, then it defeats the point of not deviating from upstream templates. I'm not willing to forego support for these Sphinx-specific templates because... well you remember #132 inquired about supporting genindex and the like.

---
> Perhaps instead of sphinx_immaterial/static, the css could be written to sphinx_immaterial/bundles 

Using sphinx_immaterial/bundles worked. I must confess, I still don't fully understand how sphinx determines what files to copy...